aea/manager/project.py

```diff
@@ -126,6 +129,22 @@ def builder(self) -> AEABuilder:
         """Get builder instance."""
         return self._get_builder(self._get_agent_config(self.path), self.path)
 
+    def check(self) -> None:
+        """Check project is ok with builder.build."""
+        builder = self.builder
+        with TemporaryDirectory() as tmp_dir:
+            key_file = str(Path(tmp_dir) / "key_file")
+            default_ledger = builder.get_default_ledger()
+            create_private_key(default_ledger, key_file)
+            if not builder.private_key_paths:
+                # no keys, but create one for builder only
+                builder.add_private_key(default_ledger, key_file)
+            if not builder.connection_private_key_paths:
+                builder.add_private_key(
+                    default_ledger, key_file, is_connection=True,
+                )
+            builder.build()
```

Comment:
I think we should just do a version check. Not sure why we do a whole build..

Reply:
the whole build is quite better, it checks many things, even python code load

Reply:
How about just this:
```
self.install_pypi_dependencies()
self.build()
```